.github/workflows/pgdg-build.yml

```diff
@@ -85,9 +85,96 @@ jobs:
       run: |
         cd pg-oidc-validator-pgdg18 && sudo tar -czvf ../pg-oidc-validator-pgdg18.tar.gz .
 
+    - name: Upload tgz artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: pg-oidc-validator-tgz
+        path: pg-oidc-validator-pgdg18.tar.gz
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    needs: [pgxs-build]
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Download built package
+      uses: actions/download-artifact@v4
+      with:
+        name: pg-oidc-validator-deb
+
+    - name: Stop default PostgreSQL
+      run: |
+        sudo systemctl stop postgresql || true
+        sudo systemctl disable postgresql || true
+
+    - name: Install PostgreSQL 18
+      run: |
+        sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt \
+          $(lsb_release -cs)-pgdg main 18" > /etc/apt/sources.list.d/pgdg.list'
+        sudo wget --quiet -O - \
+          https://www.postgresql.org/media/keys/ACCC4CF8.asc | \
+          sudo apt-key add -
+        sudo apt update
+        sudo apt -y install postgresql-18 libpq-oauth
+
+    - name: Install pg_oidc_validator package
+      run: |
+        sudo dpkg -i pg-oidc-validator-pgdg18.deb
+
+    - name: Install test dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y ruby-full ruby-bundler chromium-browser chromium-chromedriver curl net-tools
+
+    - name: Set up Ruby gems
+      run: |
+        cd test
+        bundle config set --local path 'vendor/bundle'
+        bundle install
+
+    - name: Install SSL certificate for psql
+      run: |
+        sudo cp test/keys/crt.pem /usr/local/share/ca-certificates/keycloak-test.crt
+        sudo update-ca-certificates
+
+    - name: Run tests
+      run: |
+        cd test
+        echo "PostgreSQL version:"
+        /usr/lib/postgresql/18/bin/postgres --version
+        echo "Port 5432 status:"
+        sudo netstat -tulpn | grep 5432 || echo "Port 5432 not in use"
+        echo "Chromium version:"
+        chromium-browser --version || echo "Chromium not found"
+        echo "ChromeDriver version:"
+        chromedriver --version || echo "ChromeDriver not found"
+        echo "Running tests..."
+        PGBIN=/usr/lib/postgresql/18/bin bundle exec rspec --format documentation
+
+    - name: Upload test logs on failure
+      if: failure()
+      uses: actions/upload-artifact@v4
+      with:
+        name: test-logs
+        path: |
+          test/postgres.log
+          test/psql_output.log
+          test/selenium_output.log
+          test/pgdata/log/*
+          /var/log/postgresql/*
+        if-no-files-found: ignore
+
+    - name: Download tgz artifact
```

Comment:
I wonder if it would not be cleaner to move the publishing to its own job and not have it as part of the tests. But probably something for a separate PR:

Reply:
yes, that makes sense, I'll do it in a separate pr.